crates/vm/levm/src/vm.rs

```diff
@@ -772,9 +789,7 @@ impl VM {
         );
 
         let new_address = match salt {
-            Some(salt) => {
-                Self::calculate_create2_address(current_call_frame.msg_sender, &code, salt)
-            }
+            Some(salt) => Self::calculate_create2_address(current_call_frame.to, &code, salt),
```

Comment:
Ditto. By the way, this is what is making the tests fail.

Reply:
Changed that [here](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1110/commits/c4f50b8540a7d2970900538e6b26074c8fa26a4c)